dashboard/app/models/contact_rollups_raw.rb

```diff
@@ -27,21 +27,35 @@ def self.extract_email_preferences
     ActiveRecord::Base.connection.execute(query)
   end
 
-  # @param source [String] Source from which we want to extract data
+  def self.extract_parent_emails
+    source_sql = <<~SQL
+      SELECT parent_email, MAX(updated_at) AS updated_at
+      FROM users
+      GROUP BY 1
+    SQL
+
+    query = extract_from_source_query(source_sql, [], 'parent_email', 'dashboard.users.parent_email')
+    ActiveRecord::Base.connection.execute(query)
+  end
+
+  # @param source [String] Source from which we want to extract data (can be a dashboard table name, or subquery)
   # @param data_columns [Array] Columns we want reshaped into a single JSON object
   # @param email_column [String] Column in source table we want to insert ino the email column
+  # @param source_name [String] Name for source (should be non-nil if using a subquery or non-dashboard table)
   # @return [String] A SQL statement to extract and reshape data from the source table.
-  def self.extract_from_source_query(source, data_columns, email_column)
+  def self.extract_from_source_query(source, data_columns, email_column, source_name=nil)
+    wrapped_source, sources_column = format_source(source, source_name)
+
     <<~SQL
       INSERT INTO #{ContactRollupsRaw.table_name} (email, sources, data, data_updated_at, created_at, updated_at)
       SELECT
         #{email_column},
-        'dashboard.#{source}' AS sources,
+        '#{sources_column}' AS sources,
```

Comment:
Let's just use `source_name` here and let the caller adds database name to source tables.